numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1144,7 +1158,17 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
     PyDataMem_FREE(mps);
     if (out != NULL && out != obj) {
         Py_INCREF(out);
-        PyArray_ResolveWritebackIfCopy(obj);
+        int success;
+        if (is_newarr) {
+            success = PyArray_CopyAnyInto(out, obj);
+        }
+        else {
+            success = PyArray_ResolveWritebackIfCopy(obj);
+        }
+        if (success < 0) {
+            Py_DECREF(obj);
+            goto fail;
```

Comment:
Just noticed: this is wrong!  At this point the code already did most/all cleanup and doing that again will cause crashes.  So you just need to do a straight `return NULL`.